include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -1040,10 +1040,26 @@ inline void format_bam::read_field(stream_view_type && stream_view, sam_tag_dict
             target[tag] = string_buffer;
             break;
         }
-        case 'H' :
+        case 'H' : // byte array, represented as null-terminated string
         {
-            // TODO
-            throw format_error{"'H' not implemented yet."};
+            std::vector<std::byte> byte_array;
+            std::byte value;
+            while (!is_char<'\0'>(*it))
+            {
+                string_buffer.clear();
+                string_buffer.push_back(*it);
+                ++it;
+
+                if (*it == '\0')
```

Comment:
I have troubling getting around the concept of `is_char<'\0'>(*it)`
So I am not sure if this should actually say `if (is_char<'\0'>(*it))`.